### PR TITLE
WIP: Add tests for session and session manager

### PR DESCRIPTION
**Pull request by arturomf94**
## Description
This PR will add tests for the session and session manager code (see [this](https://github.com/OpenMined/SyMPC/issues/36) for more information about the split). 

---

Closes: https://github.com/OpenMined/SyMPC/issues/39

---

## Affected Dependencies
N/A

## How has this been tested?
- Local tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by arturomf94** [comment]
@gmuraru - since I'm not sure how this split between Session and Session Manager is to be done yet, I've added a preliminary set of tests for the Session class. 

I'm not sure about the following points though: 

* In the [test for `_setup_przs`](https://github.com/OpenMined/SyMPC/blob/8c01187eee1bf6775aa6fc4e5dcc18f04773f30e/tests/sympc/session/session_test.py#L87) I'm not sure how to test such method. I've left it with `assert True` for the moment. Any ideas?

* In [this line](https://github.com/OpenMined/SyMPC/blob/8c01187eee1bf6775aa6fc4e5dcc18f04773f30e/src/sympc/session/session.py#L232) a Session instance is tested against another Session instance to see if they're equal. This seems to me as unreachable code: how could two Session instances have different `__slots__`?

PS: The MPC tensor tests seem to be failing. I found it very weird. Locally, they run OK.

**Comment by arturomf94** [comment]
Actually! Never mind! Locally they fail too :confused: @aanurraj any ideas?

**Comment by aanurraj** [comment]
> Actually! Never mind! Locally they fail too 😕 @aanurraj any ideas?

there was an issue with load_lib may be rebasing will do the work...

**Comment by arturomf94** [comment]
> > Actually! Never mind! Locally they fail too confused @aanurraj any ideas?
> 
> there was an issue with load_lib may be rebasing will do the work...

I merged `main` yesterday and it's failing. What's even more weird. Tests are failing for me on `main`. Do they run ok for you?

**Comment by arturomf94** [comment]
I just re-ran @gmuraru 's latest commit to main and indeed they fail: see [this](https://github.com/OpenMined/SyMPC/runs/1702486924).

File: src/sympc/session/session.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def _setup_przs(session: "Session") -> None:
         2 parties will generate the same random number at a given moment:
         - Party 1 holds G1 and G2
         - Party 2 holds G2 and G3
-        - Party 3 holds G3 and
+        - Party 3 holds G3 and G1
```
Comment (gmuraru) [inline src/sympc/session/session.py:191]: Thank you! :D

File: tests/sympc/session/session_test.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def test_setup_przs(clients):
+    """
+    Test _setup_przs method for session.
+    """
+    alice_client, bob_client = clients
+    session = Session(parties=[alice_client, bob_client])
+    Session._setup_przs(session)
+    assert True
```
Comment (LaRiffle) [inline tests/sympc/session/session_test.py:87]: hmmm maybe we can improve this test :D 
You could verify that `session.przs_generators` are updated as needed

File: tests/sympc/session/session_test.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+def test_przs_generate_random_share():
+    """
+    Test przs_generate_random_share method from Session.
+    """
+    session = Session()
+    generators = [torch.Generator(), torch.Generator()]
+    share = session.przs_generate_random_share(shape=(1, 2), generators=generators)
+    assert isinstance(share, ShareTensor)
+    assert (share.tensor == torch.tensor([0, 0])).all()
```
Comment (LaRiffle) [inline tests/sympc/session/session_test.py:66]: Why is this 0 ?

**Review by LaRiffle — COMMENTED** [review]
Great job!
2 small questions :)

File: tests/sympc/session/session_test.py (hunk 3 above)
Comment (arturomf94) [reply tests/sympc/session/session_test.py:66]: Good point. I think it probably shouldn't be! And I left it as it is. I'll make sure to add more tests, to make it more robust.

File: tests/sympc/session/session_test.py (hunk 2 above)
Comment (arturomf94) [reply tests/sympc/session/session_test.py:87]: Yeah, definitely needs improving. I just didn't know exactly what could be done. I'll try updates! Thanks for that :)

**Comment by arturomf94** [comment]
Closing this... Tests will be included in https://github.com/OpenMined/SyMPC/pull/44.
